tests/unit/s2n_encrypted_extensions_test.c

```diff
@@ -172,5 +172,74 @@ int main(int argc, char **argv)
         }
     }
 
+    /* Functional: Unencrypted EncryptedExtensions rejected */
+    {
```

Comment:
Should I be seeing something in here that verifies records after the handshake are ApplicationData?

Reply:
We're focused on what happens if the record we receive ISN'T an encrypted ApplicationData record.